### PR TITLE
Wish list

### PR DESCRIPTION
**Pull request by ayanjamil**
tested Wish list page on Pixel 7 ( Screen Size = 6.31 inch ) , Pixel 3a( Screen Size = 5.6 inch) , Pixel 2(Screen Size = 5.0 inch) and Pixel 7 Pro ( Screen Size = 6.71 inch)
Screenshot from 2024-01-18 14-52-53
Screenshot from 2024-01-18 14-46-11
e)
Screenshot from 2024-01-18 14-36-39

### CODE REVIEW

**Comment by udaybansal19** [comment]
@ayanjamil you can check in the screenshot that the distance between the product image and the view button is different on different screens. It should be constant across screens according to design

File: components/CardWishList.js [hunk 1]
```diff
@@ -73,73 +65,71 @@ CardWishList.propTypes = {
...
   },
-  CardWishListTitle: {
-    fontSize:16,
-    flex: 1,
-    marginTop:0,
-    marginLeft:10,
+  buttons: { 
+    flexDirection: "row",
+    alignItems: "center",  
+    marginTop:'17%',
```
Comment (udaybansal19) [inline components/CardWishList.js]: margins should me absolute

File: components/CardWishList.js [hunk 2]
```diff
@@ -73,73 +65,71 @@ CardWishList.propTypes = {
...
   },
-  CardWishListDescription: {
-    alignContent:'center',
-    padding:3,
-    paddingTop:5,
-    paddingBottom:15,
-    marginRight:60,  
+  viewBtn: {
+    backgroundColor:'#DD4A65',
+    borderRadius:20,
```
Comment (udaybansal19) [inline components/CardWishList.js]: incorrect linting

File: components/CardWishList.js [hunk 3]
```diff
@@ -47,23 +30,32 @@ class CardWishList extends React.Component {
...
               {this.renderStoreInfo()}
-              {this.renderButton()}
-            </Block>                              
+              <Block flex style={styles.buttons}>
+                <Button flex style={styles.viewBtn} onPress={() => navigation.navigate('Pro')}>
+                  View
+                </Button> 
+                <TouchableWithoutFeedback onPress={() => navigation.navigate('Pro')}>
+                  <Block flex >
+                    <Image source={require("../assets/imgs/bin.png")} style={styles.logo} />
```
Comment (udaybansal19) [inline components/CardWishList.js]: check for trash icon in icon libraries

File: screens/WishList.js [hunk 4]
```diff
@@ -5,41 +5,26 @@ import {
...
-        style={{ zIndex: 3 }}
-        key={`item-${item.title}`}
-        onPress={() => navigation.navigate("Pro", { product: item })}
-      >
-      </TouchableWithoutFeedback>
-    );
-  };
-
   renderCards = () => {
     var wishlistCards = [];
```
Comment (udaybansal19) [inline screens/WishList.js]: create a constant array and loop over the elements using .map.

**Comment by ayanjamil** [comment]
•  Top Navigation bar 
• Counter  for the number of wishlist items
• Formated code 
• Fixed padding and margin values 
• Added usable image files in constants and used in the home screen.
Screenshot from 2024-01-21 19-29-00
Screenshot from 2024-01-21 19-21-57


Screenshot from 2024-01-21 19-29-55

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "editor.defaultFormatter": "esbenp.prettier-vscode",
```
Comment (udaybansal19) [inline .vscode/settings.json]: this file shouldn't not be committed. add it to .gitignore if needed

File: assets/imgs/def.png
Comment (udaybansal19) [inline assets/imgs/def.png:1]: these files can be removed

File: components/CardWishList.js [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+import { FontAwesome } from "@expo/vector-icons";
+
+class CardWishList extends React.Component {
+  renderStoreInfo = () => {
+    return (
+      <Block flex style={styles.store}>
+        <Block>
+          <Image
+            source={{
+              uri: "https://encrypted-tbn2.gstatic.com/favicon-tbn?q=tbn:ANd9GcSrR7Ml_19H_B-zDu_Z_bQZGw0QqQ9U4wrcsyIsLYTUdUbRnR8dQRl2hFbwwMTnrsx4dn8RSieDAX0xgOMr90CKQT2BqGauj6jT9xLf-SQOorNViw",
```
Comment (udaybansal19) [inline components/CardWishList.js]: this should be sent throught props

File: components/CardWishList.js [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+            }}
+            style={styles.storelogo}
+          />
+        </Block>
+        <Block>
+          <Text size={15} style={{ color: "#A6A6A6", marginHorizontal: "7%" }}>
+            Mango
```
Comment (udaybansal19) [inline components/CardWishList.js]: this as well

File: components/CardWishList.js [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+                onPress={() => navigation.navigate("Pro")}
+              >
+                View
+              </Button>
+              <TouchableWithoutFeedback
+                onPress={() => navigation.navigate("Pro")}
+              >
+                <Block flex style={styles.trash}>
+                  {/* <Image
+                    source={require("../assets/imgs/bin.png")}
```
Comment (udaybansal19) [inline components/CardWishList.js]: please remove unused code

File: components/Header.js [hunk 9]
```diff
@@ -163,83 +264,87 @@ class Header extends React.Component {
...
-            styles.title,
-            { color: argonTheme.COLORS[white ? 'WHITE' : 'HEADER'] },
-            titleColor && { color: titleColor }
-          ]}
-          {...props}
-        />
-        {this.renderHeader()}
-      </Block>
-    );
+    if (title != "home") {
```
Comment (udaybansal19) [inline components/Header.js]: why is this check needed?

**Review by udaybansal19 — CHANGES_REQUESTED** [review]

File: screens/WishList.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    this.state = {
+      itemCount: 0,
+    };
+  }
+  componentDidMount() {
+    this.setState({ itemCount: 23 });
+  }
+  renderCards = () => {
+    var wishlistCards = [];
+    for (let i = 0; i < this.state.itemCount; i++) {
```
Comment (udaybansal19) [inline screens/WishList.js]: use `usables.map` to display all items from constants array

File: screens/WishList.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+const { width, height } = Dimensions.get("screen");
+class WishList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      itemCount: 0,
+    };
+  }
+  componentDidMount() {
+    this.setState({ itemCount: 23 });
```
Comment (udaybansal19) [inline screens/WishList.js]: this is not needed now

**Review by udaybansal19 — APPROVED** [review]

**Pull request merged**